### PR TITLE
feat: add embedded-resource-block to the list of supported block elements

### PR DESCRIPTION
**Pull request by msieroslawska**
We're working on adding resource links inside Rich Text, this PR adds the new node type for embedded entries as the first supported link type.

### CODE REVIEW

File: packages/rich-text-types/src/nodeTypes.ts [hunk 1]
```diff
@@ -94,6 +94,14 @@ export interface Link<T extends string = string> {
   };
 }
 
+export interface ResourceLink {
+  sys: {
+    type: 'ResourceLink';
+    linkType: 'Contentful:Entry';
+    urn: string;
+  };
+}
```
Comment (ghepting) [inline packages/rich-text-types/src/nodeTypes.ts:103]: our schema for third-party resources might differ slightly from this, but TBD. We'll want to make sure this interface supports both external Contentful and external 3P resources, for sure!

File: packages/rich-text-types/src/blocks.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export enum BLOCKS {
 
   EMBEDDED_ENTRY = 'embedded-entry-block',
   EMBEDDED_ASSET = 'embedded-asset-block',
+  EMBEDDED_RESOURCE = 'embedded-resource-block',
```
Comment (ghepting) [inline packages/rich-text-types/src/blocks.ts:24]: I actually wonder about this name. it's very generic, but the type interfaces being defined that represent it are very Contentful:Entry specific. This feels like enough of a future-impacting risk that we should address this now before this lands.

File: packages/rich-text-types/src/nodeTypes.ts (hunk 1 above)
Comment (anho) [reply packages/rich-text-types/src/nodeTypes.ts:103]: We can turn this into a type union in the future. Right now I don't see a blocker.

File: packages/rich-text-types/src/blocks.ts (hunk 2 above)
Comment (msieroslawska) [reply packages/rich-text-types/src/blocks.ts:24]: For the value: `embedded-resource-block` follows the names proposed in the Rich Text CEP (`embedded-resource-block`, `embedded-resource-inline` and `resource-hyperlink`). If you mean the key here, I'm open to renaming suggestions.

**Review by DanweDE — APPROVED** [review]
I noticed that the new node type is not part of `LIST_ITEM_BLOCKS`, so it couldn't be used as a list item, while entry/asset embeds can. Is that intended?

Different behavior to the other embeds might make implementation in the editor slightly more complex, although, not entirely sure how we currently enforce this behavior in the editor.

**Comment by msieroslawska** [comment]
@DanweDE that was not on purpose, thanks for catching that! I updated the schemas.

File: packages/rich-text-types/src/nodeTypes.ts (hunk 1 above)
Comment (anho) [reply packages/rich-text-types/src/nodeTypes.ts:103]: We are trying to create alignment on the topic of how to shape those 3rd party links outside of this PR. After that we will make sure to amend the schema asap.

**Pull request merged**